internal/relayenv/env_context_impl.go

```diff
@@ -73,38 +78,43 @@ type EnvContextImplParams struct {
 	UserAgent                     string
 	LogNameMode                   LogNameMode
 	Loggers                       ldlog.Loggers
+	TimeSource                    func() time.Time
+	ConnectionMapper              ConnectionMapper
 }
 
 type envContextImpl struct {
-	mu               sync.RWMutex
-	clients          map[config.SDKKey]sdks.LDClientContext
-	storeAdapter     *store.SSERelayDataStoreAdapter
-	loggers          ldlog.Loggers
-	credentials      map[credential.SDKCredential]bool // true if not deprecated
-	identifiers      EnvIdentifiers
-	secureMode       bool
-	envStreams       *streams.EnvStreams
-	streamProviders  []streams.StreamProvider
-	handlers         map[streams.StreamProvider]map[credential.SDKCredential]http.Handler
-	jsContext        JSClientContext
-	evaluator        ldeval.Evaluator
-	eventDispatcher  *events.EventDispatcher
-	bigSegmentSync   bigsegments.BigSegmentSynchronizer
-	bigSegmentStore  bigsegments.BigSegmentStore
-	bigSegmentsExist bool
-	sdkBigSegments   *ldstoreimpl.BigSegmentStoreWrapper
-	sdkConfig        ld.Config
-	sdkClientFactory sdks.ClientFactoryFunc
-	sdkInitTimeout   time.Duration
-	metricsManager   *metrics.Manager
-	metricsEnv       *metrics.EnvironmentManager
-	metricsEventPub  events.EventPublisher
-	dataStoreInfo    sdks.DataStoreEnvironmentInfo
-	globalLoggers    ldlog.Loggers
-	ttl              time.Duration
-	initErr          error
-	creationTime     time.Time
-	filterKey        config.FilterKey
+	mu                        sync.RWMutex
+	clients                   map[config.SDKKey]sdks.LDClientContext
+	storeAdapter              *store.SSERelayDataStoreAdapter
+	loggers                   ldlog.Loggers
+	identifiers               EnvIdentifiers
+	secureMode                bool
+	envStreams                *streams.EnvStreams
+	streamProviders           []streams.StreamProvider
+	handlers                  map[streams.StreamProvider]map[credential.SDKCredential]http.Handler
+	jsContext                 JSClientContext
+	evaluator                 ldeval.Evaluator
+	eventDispatcher           *events.EventDispatcher
+	bigSegmentSync            bigsegments.BigSegmentSynchronizer
+	bigSegmentStore           bigsegments.BigSegmentStore
+	bigSegmentsExist          bool
+	sdkBigSegments            *ldstoreimpl.BigSegmentStoreWrapper
+	sdkConfig                 ld.Config
+	sdkClientFactory          sdks.ClientFactoryFunc
+	sdkInitTimeout            time.Duration
+	metricsManager            *metrics.Manager
+	metricsEnv                *metrics.EnvironmentManager
+	metricsEventPub           events.EventPublisher
+	dataStoreInfo             sdks.DataStoreEnvironmentInfo
+	globalLoggers             ldlog.Loggers
+	ttl                       time.Duration
+	initErr                   error
+	creationTime              time.Time
+	filterKey                 config.FilterKey
```

Comment:
The diff here is adding:
- keyRotator
- stop/doneMonitoringCredentials
- connectionMapper (which is really just the Relay struct under the hood, needed a way to execute some callbacks on that struct)